persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -1311,6 +1313,27 @@ repsertManySql ent n = putManySql' conflictColumns fields ent n
     fields = keyAndEntityFields ent
     conflictColumns = escape . fieldDB <$> entityKeyFields ent
 
+insertUniqueSql :: EntityDef -> [PersistValue] -> InsertSqlResult
+insertUniqueSql ent vals =
+    let sql = T.concat
+            [ "INSERT INTO "
+            , escape $ entityDB ent
+            , if null (entityFields ent)
+                then " DEFAULT VALUES"
+                else T.concat
+                    [ "("
+                    , T.intercalate "," $ map (escape . fieldDB) $ entityFields ent
+                    , ") VALUES("
+                    , T.intercalate "," $ map (const "?") $ entityFields ent
+                    , ") ON CONFLICT ("
+                    , T.intercalate "," $ concat $ map (\x -> map escape (map snd $ uniqueFields x)) (entityUniques ent)
+                    , ") DO NOTHING"
+                    ]
+            ]
```

Comment:
This code is failing the following test:

```haskell
  it "insertUniqueEntity" $ runDb $ do
    let fo = Fo 3 5
    Just (Entity _ insertedFoValue) <- insertUniqueEntity fo
    Nothing <- insertUniqueEntity fo
    fo @== insertedFoValue
```

Because we want to return `Nothing` if the record was already present, while `ON CONFLICT DO NOTHING` will have the code proceed without error.